src/components/ListItem/ListItem.tsx

```diff
@@ -108,21 +109,22 @@ const ListItem: FC<ListItemProps> & { sizes?: typeof SIZES } = forwardRef(
 
     return (
       <Text
-        element="div"
-        data-testid={dataTestId || getTestId(ComponentDefaultTestId.LIST_ITEM, id)}
+        element="li"
+        data-testid={dataTestId || id}
         ref={mergedRef}
         className={cx(styles.listItem, className, getStyle(styles, camelCase(size)), {
           [styles.selected]: selected && !disabled,
           [styles.disabled]: disabled
         })}
         id={id}
+        key={id}
```

Comment:
try moving it to the `List` component

Reply:
Eventually removed it completely as I don't see this warning anymore and Text component seems like it's not passing this key anyway